File: week1/day5/nonrepeating.js
// Write a function that takes in a string of lowercase letters
// and returns the index of the string's first non-repeating character.
// If the input string does not have any non-repeating characters,
// your function should return -1.

function firstNonRepeating(string) {
  let hash = {};
  for (const letter of string) {
    if (!(letter in hash)) hash[letter] = 0;
    hash[letter]++;
  }
  for (const letter of string) {
    if (hash[letter] === 1) return `${letter},${string.indexOf(letter)}`;
  }
  return -1;
}

// returns 0
console.log(firstNonRepeating("abcdef"));
// returns 1
console.log(firstNonRepeating("abcdeaf"));
// returns 3
console.log(firstNonRepeating("bbbc"));

File: week1/day5/nonrepeating.py
def firstNonRepeating(string):
  hash = {};
  for letter in string:
    if letter not in hash:
        hash[letter] = 0;
    hash[letter]+= 1;
  
  for  letter in string:
    if hash[letter] == 1:
         return f"{letter},{string.index(letter)}";
  return -1;

# returns a, 0
print(firstNonRepeating("abcdef"));
# // returns b, 1
print(firstNonRepeating("abcdeaf"));
# // returns c, 3
print(firstNonRepeating("bbbc"));


# def repeating(string):
#     counter = {}
    
#     for char in string:
#         counter[char] = counter.get(char,0) + 1
#     for char in counter:
#         if counter[char] == 1: 
#             index = list(string).index(char)
#             print(index)
#             return print (char + ' is the first nonrepeating character')
#     return print('-1')

#     # returns a, 0
# repeating("abcdef");
# # // returns b, 1
# repeating("abcdeaf");
# # // returns c, 3
# repeating("bbbc");
